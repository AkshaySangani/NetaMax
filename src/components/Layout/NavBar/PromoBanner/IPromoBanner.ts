export interface IPromoBanner {
  /**
   * The banner text.
   * @type {string}
   *
   * */
  bannerText: string;

  /**
   * The banner text without html tags.
   * @type {string}
   *
   * */
  bannerPlainText?: string;
}
