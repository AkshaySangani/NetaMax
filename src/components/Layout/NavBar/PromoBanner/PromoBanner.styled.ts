import styled from "styled-components";
import { responsive } from "styled/responsive";

import { Box, Flex } from "@chakra-ui/react";

export const ImageContainer = styled(Box)`
  display: flex;
  align-items: center;
  justify-content: end;

  ${responsive.mdPlus} {
  }
`;

export const TextContainer = styled(Flex)`
  align-self: center;
  margin: 4px;

  ${responsive.mdPlus} {
    width: 100%;
  }
`;

export const ImageSize = styled(Flex)`
  width: 65px;
  height: 65px;
  display: flex;
  align-self: center;

  ${responsive.mdPlus} {
    display: flex;
    align-self: center;
    width: 49px;
    height: 49px;
  }
`;

export const MainContainer = styled(Box)`
  padding: 20px 24px;
  max-height: 100px;

  ${responsive.mdPlus} {
    padding: 0px;
    max-height: 49px;
  }
`;

export const BannerContainer = styled(Box)`
  ${responsive.mdPlus} {
    border-radius: 0px;
  }
`;
