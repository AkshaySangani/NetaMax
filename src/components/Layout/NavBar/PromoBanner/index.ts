// export type { IInfoHeaderProps } from "./IInfoHeaderProps";
export { PromoBanner } from "./PromoBanner";
