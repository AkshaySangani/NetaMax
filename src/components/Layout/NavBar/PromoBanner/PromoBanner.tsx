import React, { ReactElement } from "react";

import { Box, Image, Text } from "@chakra-ui/react";

import { IPromoBanner } from "./IPromoBanner";
import {
  BannerContainer,
  ImageContainer,
  ImageSize,
  MainContainer,
  TextContainer,
} from "./PromoBanner.styled";

/**
 * The PromoBanner Component
 * @param {IPromoBanner} props the PromoBanner props.
 * @returns {ReactElement} the PromoBanner component.
 */
export const PromoBanner = (props: IPromoBanner): ReactElement => {
  const { bannerPlainText } = props;
  return (
    <MainContainer>
      <BannerContainer
        backgroundColor="#5438FF"
        boxShadow="inset 0px 4px 14px rgba(0, 0, 0, 0.25)"
        borderRadius="12px"
        display="flex"
        justifyContent="center"
      >
        <Box display="flex" justifyContent="center">
          <ImageContainer>
            <ImageSize>
              <Image src="/gif/bell.gif" alt="Animate Bell" />
            </ImageSize>
          </ImageContainer>
          <TextContainer>
            <Text fontSize="13px" color="white">
              <b>{bannerPlainText || ""}</b>
            </Text>
          </TextContainer>
        </Box>
      </BannerContainer>
    </MainContainer>
  );
};
