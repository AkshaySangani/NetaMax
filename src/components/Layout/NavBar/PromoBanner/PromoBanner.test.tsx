import { render, cleanup } from "@testing-library/react";
import { PromoBanner } from "./PromoBanner";
import { IPromoBanner } from "./IPromoBanner";

describe("PromoBanner Component", () => {
  let mock: IPromoBanner;

  beforeEach(() => {
    mock = {
      bannerText: "Promo banner",
      bannerPlainText: "A demo banner",
    };
  });

  afterEach(() => {
    cleanup();
  });

  test("PromoBanner should be render correctly", () => {
    const { container } = render(<PromoBanner {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test("PromoBanner should be render correctly without plaintext", () => {
    mock = {
      bannerText: "Promo banner",
    };
    const { container } = render(<PromoBanner {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
