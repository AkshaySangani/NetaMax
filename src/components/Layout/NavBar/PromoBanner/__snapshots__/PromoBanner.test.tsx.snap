// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromoBanner Component PromoBanner should be render correctly 1`] = `
<div>
  <div
    class="sc-hKwDye bocIKv css-0"
  >
    <div
      class="sc-eCImPb jLjuHf css-4jxr00"
    >
      <div
        class="css-1l4w6pd"
      >
        <div
          class="sc-bdvvtL gZSwSj css-0"
        >
          <div
            class="sc-dkPtRN ldTtVc css-k008qs"
          >
            <img
              alt="Animate Bell"
              class="chakra-image css-0"
              src="/gif/bell.gif"
            />
          </div>
        </div>
        <div
          class="sc-gsDKAQ khYWvk css-k008qs"
        >
          <p
            class="chakra-text css-1owyea7"
          >
            <b>
              A demo banner
            </b>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PromoBanner Component PromoBanner should be render correctly without plaintext 1`] = `
<div>
  <div
    class="sc-hKwDye bocIKv css-0"
  >
    <div
      class="sc-eCImPb jLjuHf css-4jxr00"
    >
      <div
        class="css-1l4w6pd"
      >
        <div
          class="sc-bdvvtL gZSwSj css-0"
        >
          <div
            class="sc-dkPtRN ldTtVc css-k008qs"
          >
            <img
              alt="Animate Bell"
              class="chakra-image css-0"
              src="/gif/bell.gif"
            />
          </div>
        </div>
        <div
          class="sc-gsDKAQ khYWvk css-k008qs"
        >
          <p
            class="chakra-text css-1owyea7"
          >
            <b />
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
